lib/common/http1client.c

```diff
@@ -80,13 +45,15 @@ static void close_client(struct st_h2o_http1client_private_t *client)
         h2o_buffer_dispose(&client->_body_buf);
     if (client->_body_buf_in_flight != NULL)
         h2o_buffer_dispose(&client->_body_buf_in_flight);
-    free(client);
+    h2o_mem_clear_pool(&client->pool);
+    free(common);
 }
 
 static void on_body_error(struct st_h2o_http1client_private_t *client, const char *errstr)
 {
+    struct st_h2o_httpclient_private_t *common = H2O_STRUCT_FROM_MEMBER(struct st_h2o_httpclient_private_t, http1, client);
```

Comment:
Because there're two types of *client* here: `st_h2o_http1client_private_t` and `st_h2o_httpclient_private_t`. Calling both `client`s got me too confused.

Reply:
What I would do here is rename `struct st_h2o_http1client_private_t *client` to `_client`, refer to `struct st_h2o_httpclient_private_t *` as `client`, and never use `_client` once we have `client`.

This is because the members of `struct st_h2o_http1client_private_t` can always be referred to as `client->super.*`.

Using two variables that have a delta of some constant is an unnecessary complication, both at the source code-level and in machine code.

Reply:
> This is because the members of struct st_h2o_http1client_private_t can always be referred to as client->super.*.

No, st_h2o_http1client_private_t can be referred as `client->http1`. One reason I introduced this variable is for consistency with `http2client.c: Please see https://github.com/h2o/h2o/blob/a697525fca5b887f92e08e508aa35da0baf31e5e/lib/common/http2client.c#L124. So if we change as you said, should I rewrite http2client.c too not to refer `stream`, and use `client->http2` style everywhere?